### PR TITLE
Add experimental `listenForMetrics` flag

### PR DESCRIPTION
**Pull request by jasonLaster**
This adds the new experimental listenForMetrics flag for settings to devtools so we can stream metrics to the console

### CODE REVIEW

File: src/protocol/thread/thread.ts [hunk 1]
```diff
@@ -36,14 +36,19 @@ import {
...
 import { defer, assert, EventEmitter, ArrayMap } from "../utils";
 import { MappedLocationCache } from "../mapped-location-cache";
 import { ValueFront } from "./value";
 import { Pause } from "./pause";
 import uniqueId from "lodash/uniqueId";
 import { repaint } from "protocol/graphics";
 
+declare global {
+  interface Window {
+    sessionMetrics: any;
```
Comment (jcmorrow) [inline src/protocol/thread/thread.ts:49]: I know we probably won't be using it much, but it seems like we could probably at least make this like a `any[] | undefined` just to get *some* help from tsc?

File: src/ui/setup/helpers.ts [hunk 2]
```diff
@@ -11,12 +11,27 @@ declare global {
...
+    dumpBasicProcessing: () => void;
     local: () => void;
     prod: () => void;
     clearIndexedDB: () => void;
   }
 }
 
+function dumpBasicProcessing() {
+  const processing = window.sessionMetrics?.filter(
+    (data: any) => data.event == "RegionBasicProcessing"
```
Comment (jcmorrow) [inline src/ui/setup/helpers.ts:23]: Triple equals for consistency? I feel like we should have a linter for that one.

File: src/ui/setup/helpers.ts [hunk 3]
```diff
@@ -11,12 +11,27 @@ declare global {
...
+  );
+
+  const firstStartTime = processing[0]?.params.timings.creationTimestamp;
+  const stats = processing.map((region: any) => ({
+    id: region.params.regionIdx,
+    startTime: region.params.timings.creationTimestamp - firstStartTime,
+    duration: region.params.timings.processingDuration,
+  }));
+  console.table(stats);
```
Comment (jcmorrow) [inline src/ui/setup/helpers.ts:32]: nice

**Review by jcmorrow — APPROVED** [review]
Couple tiny suggestions, looks like fun 🎢

**Comment by jaril** [comment]
@jasonLaster are we still trying to get this in?

**Comment by jaril** [comment]
@jasonLaster are we still trying to get this in?

**Pull request merged**